### PR TITLE
dev-python/lxml: disable broken test-case, EAPI bump

### PR DESCRIPTION
**Pull request by MathyV**
There is a test-case not working because of issues in
libxml2. The patch disables the one broken test-case
until upstream can fix it.

Upstream bug: https://bugs.launchpad.net/lxml/+bug/1608479

Gentoo-Bug: https://bugs.gentoo.org/show_bug.cgi?id=592860

### CODE REVIEW

File: dev-python/lxml/lxml-3.4.4-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+	"${EPYTHON}" test.py -vv --all-levels -p || die "Test ${test} fails with ${EPYTHON}"
+}
+
+python_install_all() {
+	if use doc; then
+		local DOCS=( *.txt doc/*.txt )
+		local HTML_DOCS=( doc/html/. )
+	fi
+	use examples && local EXAMPLES=( samples/. )
```
Comment (SoapGentoo) [inline dev-python/lxml/lxml-3.4.4-r1.ebuild]: `EXAMPLES` is deprecated, `docinto examples` + `dodoc -r`, which you'll need to put in a proper if block

File: dev-python/lxml/lxml-3.5.0-r1.ebuild [hunk 2]
```diff
@@ -51,7 +59,7 @@ python_test() {
 	cp -r -l src/lxml/html/tests "${BUILD_DIR}"/lib/lxml/html/ || die
 	ln -s "${S}"/doc "${BUILD_DIR}"/ || die
 
-	"${PYTHON}" test.py -vv --all-levels -p || die "Test ${test} fails with ${EPYTHON}"
+	"${EPYTHON}" test.py -vv --all-levels -p || die "Test ${test} fails with ${EPYTHON}"
 }
 
 python_install_all() {
```
Comment (SoapGentoo) [inline dev-python/lxml/lxml-3.5.0-r1.ebuild]: same with `EXAMPLES` in this ebuild too

File: dev-python/lxml/lxml-3.6.4-r1.ebuild [hunk 3]
```diff
@@ -51,7 +59,7 @@ python_test() {
 	cp -r -l src/lxml/html/tests "${BUILD_DIR}"/lib/lxml/html/ || die
 	ln -s "${S}"/doc "${BUILD_DIR}"/ || die
 
-	"${PYTHON}" test.py -vv --all-levels -p || die "Test ${test} fails with ${EPYTHON}"
+	"${EPYTHON}" test.py -vv --all-levels -p || die "Test ${test} fails with ${EPYTHON}"
 }
 
 python_install_all() {
```
Comment (SoapGentoo) [inline dev-python/lxml/lxml-3.6.4-r1.ebuild]: `EXAMPLES` here

**Comment by MathyV** [comment]
I changed the EXAMPLES, also restored 3.6.0 as I overlooked it had some stable keywords which QA bot didn't like.

File: dev-python/lxml/lxml-3.4.4-r1.ebuild [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+python_prepare_all() {
+	# avoid replacing PYTHONPATH in tests.
+	sed -i '/sys\.path/d' test.py || die
+
+	distutils-r1_python_prepare_all
+}
+
+python_compile() {
+	if [[ ${EPYTHON} != python3* ]]; then
+		local CFLAGS=${CFLAGS}
```
Comment (SoapGentoo) [inline dev-python/lxml/lxml-3.4.4-r1.ebuild]: make this `local -x` instead, and do so in the other ebuilds too

**Comment by MathyV** [comment]
Done

**Pull request closed without merging**